### PR TITLE
Reworking teams, adding force-pause feature, adding pause/resume messages :

### PR DESCRIPTION
**Pull request by ravomavain**
## Reworking teams to allow pausing in team
- Adding wrappers functions to modify character or pause data if player are in game or paused.
- Moved OnFinish() function to teams.cpp so it can also be called on paused players.
- Pause is now enabled for team players.
- Adding bool CPlayer::IsPlaying() function that return true if the player is either in game or in pause. (false if spectator/dead).
## Adding vote to force-pause players (instead of just moving them to spectators).
- Add the following config variables :
  - sv_vote_pause: Allow voting to pause players (instead of moving to spectators) (default: 1)
  - sv_vote_pause_time: The time (in seconds) players have to wait in pause when paused by vote (default: 6
  - sv_vote_pause_auto: Automatically unpaused players after the force-pause delay or not (default: 0)
- Add the following functions :
  - force_pause v?i : Force v to pause for i seconds
  - force_unpause v : Force v to unpause

### CODE REVIEW

**Comment by heinrich5991** [comment]
I dislike the force_pause, force_unpause because it's easy to annoy the player with it (for example: an admin and his noobfriends do a map, they want to get past the other tees, pausing them, they won't leave because they'll see a chance coming back. such an admin wouldn't probably kick the players, and if so, they wouldn't rejoin, so the annoyance factor isn't that big).
I also dislike sv_vote_pause_time, because this speccing (and also pausing) players is actually ment to be used on afk players only, and if a player is moved to spectator/paused because he is annoying, he has two options: rejoin, restarting annoying and getting kickvoted or rejoin, stopping annoying and not getting kickvoted. So I'd like to get the time removed from this pausevote.

**Comment by ravomavain** [comment]
I disagree about the sv_vote_pause_time : voting are hard and long to past, so when someone get force_pause it have to be more than just moving him for a very short time : this is not only to be used for afk players (which are not really annoying since they are afk), it's for players who don't want to let other try a part where he fail for example, in this case having a join delay let other tee try the part without being board by other players.
And I hate to kick players : it's really rude, especially on the middle of a map, even if a player is sometime annoying you, it's not a reason to kick him, force_pausing will be a kind of warning for him before kicking.
But default sv_vote_pause_time should be shorter (what about 10s)

About force_pause and force_unpause commands : 
force_pause is required for the vote (like kick and set_team commands for kicking and moving to spec) ! And even if we set it to a higher level (CONSOLELEVEL_CONFIG), admin can still force_vote to force_pause players. I can't do anything against noobish-admins stupidness !
The force_unpause command is in fact quiet useless, since we can use "force_pause v 0" to cancel force_pause. (which is something required for abusiv pause votes)

For afk players, if they are automatically force_paused, it will be without time (ie force_pause v 0)

**Comment by heinrich5991** [comment]
I disagree again :)

See the pause vote as the first warning, if the blocker sees that it passes, he'll probably not restart blocking, but even if, then you can kickvote him. (See also my last post about that)

"I can't do anything against noobish-admins stupidness !"
That's the very wrong approach in my opinion, you should simply :) create options that are least abusable.

File: src/game/server/ddracecommands.cpp [hunk 1]
```diff
@@ -466,26 +467,35 @@ void CGameContext::ConTogglePause(IConsole::IResult *pResult, void *pUserData, i
...
 				pPlayer->SaveCharacter();
-				pPlayer->SetTeam(TEAM_SPECTATORS);
 				pPlayer->m_InfoSaved = true;
+				pPlayer->SetTeam(TEAM_SPECTATORS);
 			}
 			else
 				pResult->Print(IConsole::OUTPUT_LEVEL_STANDARD, "info", "You can\'t pause that often.");
 		}
-		else if(pPlayer->GetTeam()==TEAM_SPECTATORS && pPlayer->m_InfoSaved)
+		else if(pPlayer->GetTeam()==TEAM_SPECTATORS && pPlayer->m_InfoSaved && pPlayer->m_ForcePauseTime==0)
```
Comment (heinrich5991) [inline src/game/server/ddracecommands.cpp]: spaces are missing around the '==' operator :)

**Comment by ravomavain** [comment]
IRL, after a trial, people are not put in jail with the keys! They remain there for some time. Here it's the same: it must bother the player, or he will do it again. And once again, kick is not a good solution imo.

And there is already many option that can be abused, like kick, set_team, tune, kill_pl, ... If there is these noobish-admins on a server, change server ! They don't deserve that you spend your time on there servers.

File: src/game/server/ddracecommands.cpp (hunk 1 above)
Comment (ravomavain) [inline src/game/server/ddracecommands.cpp]: Sir, yes, sir!

**Pull request closed without merging**